hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java

```diff
@@ -825,6 +827,20 @@ private void createRSGroupTable() throws IOException {
     }
 
     public boolean isOnline() {
```

Comment:
How often is this check run?

Reply:
@saintstack it's called by `RSGroupBasedLoadBalancer#balanceCluster` ([here])(https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupBasedLoadBalancer.java#L114), `RSGroupInfoManagerImpl#refresh` ([here](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java#L581)), and `RSGroupInfoManagerImpl#flushConfig` ([here](https://github.com/apache/hbase/blob/master/hbase-server/src/main/java/org/apache/hadoop/hbase/rsgroup/RSGroupInfoManagerImpl.java#L635)) so I would think pretty frequently. Not sure how often `balanceCluster` gets called, but `refresh` gets called upon `RSGroupInfoManagerImpl` startup and `flushConfig` gets called every time we add/remove servers or tables or change the rsgroups in any way.